device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala

```diff
@@ -0,0 +1,48 @@
+package org.genivi.sota.device_registry
+
+import slick.driver.MySQLDriver.api._
+import akka.http.scaladsl.marshalling.Marshaller._
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.{Directive1, Directives, Route}
+import io.circe.Json
+import io.circe.generic.auto._
+import org.genivi.sota.data.Namespace
+import org.genivi.sota.marshalling.RefinedMarshallingSupport._
+import org.genivi.sota.device_registry.common.CreateGroupRequest
+
+import scala.concurrent.ExecutionContext
+
+class GroupsResource(namespaceExtractor: Directive1[Namespace])
+                    (implicit ec: ExecutionContext,
+                    db: Database)
+  extends Directives {
+
+  import org.genivi.sota.marshalling.CirceMarshallingSupport._
+
+  def createGroupFromDevices(request: CreateGroupRequest, namespace: Namespace): Route = {
+    val res = for {
+      info1 <- db.run(SystemInfo.findByUuid(request.device1))
+      info2 <- db.run(SystemInfo.findByUuid(request.device2))
+    } yield JsonComparison.getCommonJson(info1, info2)
+
+    onSuccess(res) {
+        case Json.Null =>
+          complete(StatusCodes.BadRequest -> "Devices have no common attributes to form a group")
+        case json =>
+          val dbIO = for {
+            _ <- GroupInfoRepository.create(request.groupName, namespace, json)
+            _ <- GroupMember.createGroup(request, namespace)
+                 //PRO-1378 Add logic to find devices which should be in this group
+          } yield ()
+          complete(db.run(dbIO.transactionally))
+    }
+  }
+
+  val route: Route =
+    (pathPrefix("device_groups") & namespaceExtractor) { ns =>
+      (post & pathPrefix("from_attributes") & pathEnd) {
```

Comment:
Yeah we could, I did it like this because I was planning to move the other group routes in here, for https://advancedtelematic.atlassian.net/browse/PRO-1463. I can keep them separate if you prefer?